pyuvdata/uvdata/mir_parser.py

```diff
@@ -225,696 +42,1062 @@ class MirParser(object):
+            with two items: "scale_fac", and np.int16 which describes the common
+            exponent for the spectrum, and "data", an array of np.int16 (of length
+            equal to twice that found in `sp_data["nch"]` for the corresponding value
+            of sphid) containing the compressed visibilities.  Note that entries equal
+            to -32768 aren't possible with the compression scheme used for MIR, and so
+            this value is used to mark flags.
 
-        # Create a temporary dictionary for filtering the remaining dtypes
-        inhid_filter_dict = {
-            key: value for key, value in zip(self.in_read["inhid"], self.in_filter)
+        Returns
+        -------
+        vis_dict : dict
+            A dictionary in the format of `vis_data`, where the keys are matched to
+            individual values of sphid in `sp_data`, and each entry comtains a dict
+            with two items: "data", an array of np.complex64 containing the
+            visibilities, and "flags", an array of bool containing the per-channel
+            flags of the spectrum (both are of length equal to `sp_data["nch"]` for the
+            corresponding value of sphid).
+        """
+        # The code here was derived after a bunch of testing, trying to find the fastest
+        # way to covert the compressed data into the "normal" format. Some notes:
+        #   1) The setup below is actually faster than ldexp, probably because of
+        #      the specific dtpye we are using.
+        #   2) Casting 2 as float32 will appropriately cast sp_raw valuse into float32
+        #      as well.
+        #   3) I only check for the "special" value for flags in the real component. A
+        #      little less robust (both real and imag are marked), but faster and
+        #      barring data corruption, this shouldn't be an issue (and a single bad
+        #      channel sneaking through is okay).
+        #   4) pairs of float32 -> complex64 is super fast and efficient.
+        vis_dict = {
+            sphid: {
+                "data": ((np.float32(2) ** sp_raw["scale_fac"]) * sp_raw["data"]).view(
+                    dtype=np.complex64
+                ),
+                "flags": sp_raw["data"][::2] == -32768,
+            }
+            for sphid, sp_raw in raw_dict.items()
         }
 
-        # Filter out the last two data products, based on the above
-        self.eng_filter = np.array(
-            [inhid_filter_dict[key] for key in self.eng_read["inhid"]]
-        )
-        self.we_filter = np.array(
-            [inhid_filter_dict[key] for key in self.we_read["scanNumber"]]
-        )
-
-        if self._has_auto:
-            self.ac_filter = np.array(
-                [inhid_filter_dict[key] for key in self.ac_read["inhid"]]
-            )
+        # In testing, flagging the bad channels out after-the-fact was significantly
+        # faster than trying to much w/ the data above.
```

Comment:
maybe: to -> too? Something's not quite right with this sentence

Reply:
Heh, this sentence was pretty butchered -- fixed in a2f46b4b.